code/modules/projectiles/guns/lawgiver.dm

```diff
@@ -155,62 +256,43 @@
 
 /obj/item/weapon/gun/lawgiver/proc/RemoveMag(var/mob/user)
 	if(magazine)
-		var/obj/item/ammo_storage/magazine/lawgiver/L = magazine
-		L.forceMove(get_turf(src.loc))
+		magazine.forceMove(get_turf(src.loc))
 		if(user)
-			user.put_in_hands(L)
+			user.put_in_hands(magazine)
 			to_chat(user, "<span class='notice'>You pull the magazine out of \the [src].</span>")
-		L.update_icon()
+		magazine.update_icon()
 		magazine = null
 		update_icon()
 		return 1
 	return 0
 
+/obj/item/weapon/gun/lawgiver/proc/activate_mode(var/datum/lawgiver_mode/mode)
+	set waitfor = FALSE
```

Comment:
There surely is a reason for this but I don't quite see it.
Is it so that the lawgiver speaks after the user ?

Reply:
yes